File: wpsSimulator/src/main/java/org/wpsim/SimulationControl/Util/SimulationParams.java
/**
 * ==========================================================================
 * __      __ _ __   ___  *    WellProdSim                                  *
 * \ \ /\ / /| '_ \ / __| *    @version 1.0                                 *
 * \ V  V / | |_) |\__ \  *    @since 2023                                  *
 * \_/\_/  | .__/ |___/   *                                                 *
 * | |                    *    @author Jairo Serrano                        *
 * |_|                    *    @author Enrique Gonzalez                     *
 * ==========================================================================
 * Social Simulator used to estimate productivity and well-being of peasant *
 * families. It is event oriented, high concurrency, heterogeneous time     *
 * management and emotional reasoning BDI.                                  *
 * ==========================================================================
 */
package org.wpsim.SimulationControl.Util;

public class SimulationParams {
    /** Container alias used as the BESA container name and RabbitMQ routing key. */
    public String mode = "single";
    /**
     * Node role: {@code primary} creates services + peasants (default),
     * {@code worker} creates only peasants (for distributed deployments).
     * Inferred from mode if not supplied explicitly.
     */
    public String role = null;
    public String env;
    public int money = -1;
    public int land = -1;
    public double personality = -1.0;
    public int tools = -1;
    public int seeds = -1;
    public int water = -1;
    public int irrigation = -1;
    public int emotions = -1;
    public int training = -1;
    public int nodes = 0;
    public int steptime = 50;
    public int years = 1;
    public String world;

}

File: ethosterra-python/ethosterra/simulation_params.py
from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any


@dataclass
class SimulationParams:
    mode: str = "single"
    role: str | None = None
    env: str = ""
    agents: int = 5
    money: int = 3000000
    land: int = -1
    personality: float = -1.0
    tools: int = -1
    seeds: int = -1
    water: int = -1
    irrigation: int = -1
    emotions: int = -1
    training: int = -1
    nodes: int = 0
    years: int = 1
    variance: float = -1.0
    criminality: int = -1
    steptime: int = -1
    perturbation: str = ""
    training_slots: int = -1
    world: str = "100"
    world_lands: int = 800
    world_file: str = ""
    speed: float = 0.001
    crime_rate: float = 0.0
